api/institutions/authentication.py

```diff
@@ -113,6 +119,14 @@ def authenticate(self, request):
                 osf_support_email=OSF_SUPPORT_EMAIL,
                 storage_flag_is_active=waffle.flag_is_active(request, 'storage_i18n'),
             )
+        elif not user.is_active:
+            # An existing user object has been found with the matching institution email.  If the
+            # user turns out to be an inactive one (e.g. unclaimed, unconfirmed, etc.), however,
+            # simply send a sentry message and return the user.  DON'T affiliate the institution.
+            # CAS will handle the rest and inform the institution user.
+            sentry.log_message('The authenticated institution user is inactive: '
+                               'user={}, inst={}'.format(username, provider['id']))
+            return user, None
 
         if not user.is_affiliated_with_institution(institution):
```

Comment:
Would it make more sense/be more explicit to just change the following to
```
if not user.is_affiliated_with_institution(institution) and user.is_active:
	user.affiliated_institutions.add(institution)
	user.save()
```
so that institutions are only affiliated with active users?